js/preview/src/dev_server.ts

```diff
@@ -27,19 +27,21 @@ interface ServerOptions {
 	root_dir: string;
 	frontend_port: number;
 	backend_port: number;
+	host: string;
 }
 
 export async function create_server({
 	component_dir,
 	root_dir,
 	frontend_port,
-	backend_port
+	backend_port,
+	host
 }: ServerOptions): Promise<void> {
 	process.env.gradio_mode = "dev";
 	const imports = generate_imports(component_dir);
 
 	const NODE_DIR = join(root_dir, "..", "..", "node", "dev");
-
+	const host_ = host == "localhost" ?  undefined : host;
```

Comment:
I think we can skip this and just pass `host` in as it is. We are setting a default value in the CIL that is equal to vite's default value, so passing it through shouldn't cause any issues.